src/marks/auto.js

```diff
@@ -104,6 +99,49 @@ export function auto(data, {x, y, color, size, fx, fy, mark} = {}) {
         : null;
   }
 
+  // TODO: Maybe we should return null in undefined cases, so the spec is always
+  // totally defined one way or the other. (Currently breaks a lot of tests.)
+  return {
+    x: {
+      ...(materialize ? {value: x} : xValue !== undefined && {value: xValue}),
+      ...(xReduce !== undefined && {reduce: xReduce}),
+      ...(xZero !== undefined && {zero: xZero}),
+      ...xOptions
+    },
+    y: {
+      ...(materialize ? {value: y} : yValue !== undefined && {value: yValue}),
+      ...(yReduce !== undefined && {reduce: yReduce}),
+      ...(yZero !== undefined && {zero: yZero}),
+      ...yOptions
+    },
+    color: {
+      ...(materialize ? {value: color} : colorValue !== undefined && {value: colorValue}),
+      ...(colorColor !== undefined && {color: colorColor}),
+      ...(colorReduce !== undefined && {reduce: colorReduce})
+    },
+    size: {
+      ...(materialize ? {value: size} : sizeValue !== undefined && {value: sizeValue}),
+      ...(sizeReduce !== undefined && {reduce: sizeReduce})
+    },
+    ...(fx !== undefined && {fx}),
+    ...(fy !== undefined && {fy}),
+    mark
+  };
+}
+
+/** @jsdoc auto */
+export function auto(data, initialOptions) {
+  let {
+    x: {value: x, reduce: xReduce, zero: xZero, ...xOptions},
+    y: {value: y, reduce: yReduce, zero: yZero, ...yOptions},
+    color: {value: color, color: colorColor, reduce: colorReduce},
+    size: {value: size, reduce: sizeReduce},
+    fx,
+    fy,
+    mark
+  } = spec(data, initialOptions, {materialize: true});
```

Comment:
A different way we can do this (without requiring an internal option and separate function) is to materialize the values as arrays, and then pass them into autoSpec; this way valueof does nothing (it returns the existing arrays). I’ll take a crack at this and push a commit if it works.